src/test/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMapTest.java

```diff
@@ -86,8 +89,7 @@ public void testAutoRefresh() throws InterruptedException {
 		Assert.assertEquals("initial", value);
 		
 		// we have to wait until the thread had a chance to refresh
-		Thread.sleep(1000);
-		
+		await().atMost(5, SECONDS).until(() -> !subject.get("key").equals("initial"));
```

Comment:
I fear this changes the semantics of the test: `subject.get()` would load the entry, if it was evicted. See also https://github.com/promregator/promregator/blob/744b4044fc424c3ba729e7c354a66f2c7c328589/src/main/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMap.java#L147 

Note that `subject.containsKey()` does not this.
However, what we want to test here

Reply:
Hopefully my updated changes do not change test semantics